src/features/TokenMigration/stateFactory.tsx

```diff
@@ -0,0 +1,22 @@
+import { TUseStateReducerFactory } from '@utils';
+import { TokenMigrationState, TStepAction, ISimpleTxFormFull } from './types';
+const TokenMigrationFactory: TUseStateReducerFactory<TokenMigrationState> = ({
+  state,
+  setState
+}) => {
+  const handleUserInputFormSubmit: TStepAction = (payload: ISimpleTxFormFull) => {
```

Comment:
We want to use `reducers` over `stateFactories`.

Reply:
I'm using the `useStateReducer` hook!

Reply:
I pushed a reducer which is more of the pattern that we want to use, please take a look!